bundles/org.openhab.binding.homekit/src/main/java/org/openhab/binding/homekit/internal/handler/HomekitAccessoryHandler.java

```diff
@@ -0,0 +1,1026 @@
+                        switch (link.cxxType) {
+                            case HUE -> {
+                                QuantityType<Angle> hue = QuantityType.valueOf(lightModel.getHue(), Units.DEGREE_ANGLE);
+                                updateState(link.channel.getUID(), hue);
+                            }
+                            case SATURATION -> {
+                                PercentType sat = new PercentType(BigDecimal.valueOf(lightModel.getSaturation()));
+                                updateState(link.channel.getUID(), sat);
+                            }
+                            case BRIGHTNESS -> {
+                                if (lightModel.getBrightness(true) instanceof PercentType bri) {
+                                    updateState(link.channel.getUID(), bri);
+                                }
+                            }
+                            case ON -> {
+                                if (lightModel.getOnOff(true) instanceof OnOffType onOff) {
+                                    updateState(link.channel.getUID(), onOff);
+                                }
+                            }
+                            default -> {
+                            }
+                        }
+                    });
+                }
+            } else {
+                writeChannel(channel, command);
+            }
+            return; // success
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt(); // shutting down; restore interrupt flag but otherwise do nothing
+        } catch (Exception e) {
+            if (isCommunicationException(e)) {
+                // communication exception; log at debug and try to reconnect
+                logger.debug("{} communication error '{}' sending command '{}' to '{}', reconnecting..", thing.getUID(),
+                        e.getMessage(), command, channelUID, e);
+                scheduleConnectionAttempt();
+            } else {
+                // other exception; log at warn and don't try to reconnect
+                logger.warn("{} unexpected error '{}' sending command '{}' to '{}'", thing.getUID(), e.getMessage(),
+                        command, channelUID, e);
+            }
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    THING_STATUS_FMT.formatted("error.error-sending-command", e.getMessage()));
+        }
+    }
+
+    @Override
+    public void initialize() {
+        super.initialize();
+        if (isBridgedAccessory) {
+            if (getBridge() instanceof Bridge bridge && bridge.getStatus() == ThingStatus.ONLINE) {
+                scheduler.submit(() -> {
+                    onConnectedThingAccessoriesLoaded();
+                    updateStatus(ThingStatus.ONLINE);
+                });
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_UNINITIALIZED);
```

Comment:
This combination is not correct, see https://www.openhab.org/docs/concepts/things.html#status-details.

`initialize` will not be called for child handlers until the bridge is no longer uninitialized (i.e. UNKNOWN, OFFLINE or ONLINE). In case the bridge is OFFLINE, BRIDGE_OFFLINE would apply.

Reply:
Ok. I will revert that.